oss_src/flexible_type/flexible_type_converter.hpp

```diff
@@ -13,621 +13,619 @@
- *
- * The key is to support as interesting types for T as possible.
- * The following are currently supported:
- *  - flexible_type itself
- *  - flexible_type castable cases:
- *     - Any direct member of flexible_type:
- *        - flex_int (int64_t)
- *        - flex_float (double)
- *        - flex_string (std::string)
- *        - flex_vec (std::vector<double>)
- *        - flex_list (std::vector<flexible_type>)
- *        - flex_dict (std::vector<std::pair<flexible_type, flexible_type> >)
- *        - flex_date_time (boost::ptime)
- *  - Any scalar type (any boolean, integer, or floating point type)
- *  - Recursive cases
- *     - std::vector<T> where T is of any type in this list including the
- *     recursive cases.  
- *     - std::map<S, T> where S and T are of any type in this list including
- *     the recursive cases.  (though note that complicated S types may not be
- *     representable in Python.)
- *     - std::unordered_map<S, T> where S and T are of any type in this list
- *     including the recursive cases.  (though note that complicated S types
- *     may not be representable in Python.)
- *     - std::pair<S, T> where S, T are of any type in this list including the
- *     recursive cases.
- *     - std::tuple<T...> where T... are of any type in this list including the
- *     recursive cases.
- * 
- * flexible_type_converter_implementation Details
- * ----------------------
- * The key difficulty is to make sure that every T matches *exactly* one case.
- * Each case below is numbered, and we document here how each situation maps
- * to each case.
- *
- * Base case. 
- * All failed templatizations will come here where compilation will fail.
+/** straight flexible_type.  Always given priority. 
  */
-template <typename T, class Enable=void>
-struct flexible_type_converter {
-  // every successful specialization has a value = True. 
-  // This one has a value = False.
-  // This value is used in is_flexible_type_convertible<T> and 
-  // all_flexible_type_convertible<Args...> to test if stuff are convertible 
-  // to flexible_type via flexible_type_converter<T>
-  static constexpr bool value = false;
+template <> struct ft_converter<1> {
+  
+  template <typename T> static constexpr bool matches() {
+    return std::is_same<T, flexible_type>::value;
+  }
+
+  template <typename T> 
+  static void get(T& dest, const flexible_type& src) {
+    dest = src;
+  }
+  
+  template <typename T> 
+  static void set(flexible_type& dest, const T& src) {
+    dest = src;
+  }
 };
 
-/**
- * Case 1a: Cover all the direct members of flexible_type, 
- *         EXCEPT for numeric types (flex_int, flex_float), and flex_list.
- *        - flex_string (std::string)
- *        - flex_vec (std::vector<double>)
- *        - flex_dict (std::vector<std::pair<flexible_type, flexible_type> >)
- *        - flex_date_time (boost::ptime)
+////////////////////////////////////////////////////////////////////////////////
+
```

Comment:
I'm not sure that's true...  These would use the `v1.assign(v2.begin(), v2.end())` methods instead of `v1 = v2`, which should be exactly the same for identical matches of vector types.  I can add it these cases for clarity, but I think it's already as fast as possible.